packages/teleport/src/DesktopSession/useDesktopSession.tsx

```diff
@@ -31,6 +31,8 @@ export default function useDesktopSession(ctx: Ctx) {
   const [hostname, setHostname] = useState<string>('');
   const clientCanvasProps = useTdpClientCanvas();
 
+  document.title = `${clusterId} • ${clientCanvasProps.username}@${hostname}`;
+
```

Comment:
Instead of `  const { clusterId, desktopId } = useParams<UrlDesktopParams>();`, could you return this 2 values from `useTdpClientCanvas` as it has the same line? Or do this instead?

```ts

const { clusterId, username, desktopId } = useParams<UrlDesktopParams>();
const tdpClientCanvas = useTdpClientCanvas({ clusterId, desktopId, username})

useEffect(() => {
  document.title = `${clusterId} • ${username}@${hostname}`;
}, [hostname])

useEffect(() => {    
  run(() => 
    // TODO: add a method to the backend that returns a desktop by ID, meanwhile move that code to the desktopService.getDesktop method. Also, there is no need to carry on `stripRdpPort(desktop.addr)` all the way from backend to the client side. This is bad API design. 
    ctx.desktopService
      .getDesktop({ clusterId, desktopId)
      .then(desktop => {    
        setHostname(desktop.hostname);
      })    
  ));
}, [clusterId, desktopId]);

```